IPython/core/display.py

```diff
@@ -594,6 +595,9 @@ def __init__(self, data=None, url=None, filename=None, metadata=None):
         metadata : dict
             Dict of metadata associated to be the object when displayed
         """
+        if isinstance(data, (Path, PurePath)):
```

Comment:
Hi there. 

Just a suggestion: you are repeating 4 times the same code here. Why don't create a separate function and use it instead? I believe this way code will be cleaner.

Reply:
That would not shorten the code that much, so I'm not sure it would be worth it for 4 instances; 

For a reader it might also interrupt the flow and have them check what this hypothetical function would do.

I believe in this case inlining the code is good enough.